modules/delegates/helpers/filterDelegates.ts

```diff
@@ -6,77 +6,52 @@ SPDX-License-Identifier: AGPL-3.0-or-later
 
 */
 
-import { DelegateStatusEnum, DelegateTypeEnum } from '../delegates.constants';
-import { AllDelegatesEntryWithName, DelegatePaginated } from '../types';
+import { DelegateTypeEnum } from '../delegates.constants';
+import { AllDelegatesEntryWithName } from '../types';
 
 export function filterDelegates(
-  delegates: DelegatePaginated[],
-  showShadow: boolean,
-  showAligned: boolean,
-  showExpired: boolean,
-  name: string | null,
-  avcs?: { [key: string]: boolean }
-): DelegatePaginated[] {
-  return (
-    delegates
-      // name filter
-      .filter(delegate => {
-        if (!name || name.trim() === '') return true;
-
-        return delegate.name.toLowerCase().includes(name.toLowerCase());
-      })
-      .filter(delegate => {
-        // filter out expired if show expired not checked
-        if (!showExpired && delegate.expired === true) {
-          return false;
-        }
-
-        // return all if show shadow and show aligned are both unchecked
-        if (!showShadow && !showAligned) {
-          return true;
-        }
-
-        if (!showShadow && delegate.status === DelegateStatusEnum.shadow) {
-          return false;
-        }
-
-        if (!showAligned && delegate.status === DelegateStatusEnum.aligned) {
-          return false;
-        }
-
-        return true;
-      })
-      // Filter by tags
-      .filter(delegate => {
-        // CVS act as a OR filter
-        if (!avcs) return true;
-
-        const avcArray = Object.keys(avcs).filter(key => avcs[key]);
-        if (avcArray.length === 0) {
-          return true;
-        }
-
-        return delegate.avc_name && avcArray.includes(delegate.avc_name);
-      })
+  allDelegatesWithNames: AllDelegatesEntryWithName[],
+  queryAvcs: string[] | null,
+  searchTerm: string | null,
+  type?: DelegateTypeEnum
+): {
+  alignedDelegatesAddresses: string[];
+  filteredDelegateAddresses: string[];
+  filteredDelegateEntries: AllDelegatesEntryWithName[];
+} {
+  const alignedDelegatesAddresses = filterDelegateAddresses(allDelegatesWithNames, null, null);
+  const filteredDelegateAddresses = filterDelegateAddresses(
+    allDelegatesWithNames,
+    queryAvcs,
+    searchTerm,
+    type
   );
+  const filteredDelegateEntries =
+    !searchTerm && !queryAvcs
+      ? allDelegatesWithNames
+      : allDelegatesWithNames.filter(delegate => filteredDelegateAddresses.includes(delegate.voteDelegate));
+
+  return { alignedDelegatesAddresses, filteredDelegateAddresses, filteredDelegateEntries };
```

Comment:
i still feel that this is a bit weird, is a filter function but it returns an unfiltered result with the addresses. I'll approve it but in the future i would advice to maybe make this code refactored in a way that is more easy to understand what it does